src/NzbDrone.Core/Notifications/Telegram/Telegram.cs

```diff
@@ -1,16 +1,78 @@
+using System;
 using System.Collections.Generic;
 using FluentValidation.Results;
 using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Tv;
 
 namespace NzbDrone.Core.Notifications.Telegram
 {
     public class Telegram : NotificationBase<TelegramSettings>
     {
+        private const string ImdbUrlFormat = "https://www.imdb.com/title/{0}";
+        private const string TvdbUrlFormat = "http://www.thetvdb.com/?tab=series&id={0}";
+        private const string TraktUrlFormat = "http://trakt.tv/search/tvdb/{0}?id_type=show";
+        private const string TvMazeUrlFormat = "http://www.tvmaze.com/shows/{0}/_";
+
         private readonly ITelegramProxy _proxy;
+        private readonly Dictionary<MetadataLinkType, Func<string, string, string>> _formatLinkFromIdMethods;
+
+        private string FormatImdbLinkFromId(string message, string id)
```

Comment:
Could this be encapsulated in a utility class somehow? Might be useful for other notication providers?

Reply:
Potentially, but it depends a lot on the format something else might need, so for now I wouldn't worry about it.

This whole group of methods can be cleaned up and combined with `FormatMessageWithLink`

Something like:

```
private string FormatMessageWithLink(string message, Series series)
{
	var linkType = (MetadataLinkType)Settings.MetadataLinkType;

	switch (linkType)
    {
        case MetadataLinkType.IMDb:
            return $"[{message}](https://www.imdb.com/title/{series.ImdbId})";
        ...
        default:
            throw new ArgumentException($"Unsupported link type: {linkType}", nameof(linkType));
    }
}

Reply:
Also move these private functions to the bottom of the class.